iree/base/api.h

```diff
@@ -653,7 +653,55 @@ typedef struct iree_status_handle_t* iree_status_t;
 // TODO(#2843): better logging of status checks.
 #define IREE_CHECK_OK(expr) \
   IREE_CHECK_EQ(IREE_STATUS_OK, iree_status_consume_code(expr))
-#define IREE_ASSERT_ARGUMENT(name) assert(name)
+
+//===----------------------------------------------------------------------===//
+// IREE_ASSERT macros
+//===----------------------------------------------------------------------===//
+// These are no-oped in builds with NDEBUG defined (by default anything but
+// `-c dbg`/`-DCMAKE_BUILD_TYPE=Debug`). As with normal assert() ensure that
+// side-effecting behavior is avoided as the expression will not be evaluated
+// when the asserts are removed!
+
+// TODO(benvanik): move to iree/base/assert.h.
+#ifndef NDEBUG
```

Comment:
I feel like there are a lot of negatives here. Can we flip the condition?

`#if defined(NDEBUG)`